threadproc/win32/thread.c

```diff
@@ -284,3 +284,13 @@ APR_DECLARE(int) apr_os_thread_equal(apr_os_thread_t tid1,
 }
 
 APR_POOL_IMPLEMENT_ACCESSOR(thread)
+
+APR_DECLARE(apr_status_t) apr_thread_setname(apr_thread_t *thread, const char *name)
+{
+	return APR_ENOTIMPL;
```

Comment:
https://msdn.microsoft.com/en-us/library/windows/desktop/mt774976(v=vs.85).aspx

Reply:
Hello @jeking3 
I now have windows PC and cannot build and test code. Could you write create PR to my repo? I will  merge your PR into this.

Reply:
I would recommend committing this because technically it isn't wrong - it returns "not implemented", and someone will do then win32 side when they need it.  I would implement it such that the documented API would be used, and then if Windows is an earlier version than that API supports, to use the trick found in https://www.codeproject.com/articles/8549/name-your-threads-in-the-vc-debugger-thread-list.

If pull requests here leveraged Travis CI and Appveyor you would be able to make those changes and see the tests pass... it's too bad the project doesn't leverage the free CI tools that exist for open source projects.

Reply:
@jeking3, could you mark PR as reviewed.

Reply:
Sure, although I should ask why folks more closely tied to the project are not reviewing the pull requests?  I am a PMC member on Thrift, and I maintain some Boost libraries... I don't look at apr that often.

Reply:
@jeking3: activity mostly happens on the dev@apr.apache.org mailing list and bugzilla.
PRs are looked at from time to time, by individuals...

@sergey-safarov: a new function mostly ENOTIMPL (but on some pthread implementations, as _np suffix suggests) is not very attractive for APR (P stands for Portable). Any chance to at least handle Windows per MSDN link? (I can't compile nor test myself unfortunately).
Maybe for your use case apr_os_thread_get() could help to do native calls on the native thread descriptor.